pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go

```diff
@@ -188,10 +189,11 @@ func (le *LeaderEngine) EnsureLeaderElectionRuns() error {
 }
 
 func (le *LeaderEngine) runLeaderElection() {
+	leCtx := context.Background()
+	go le.reportLeaderMetric(leCtx)
```

Comment:
maybe we can generate/update the metrics in this other goroutine: https://github.com/DataDog/datadog-agent/blob/96195fbd565bd5120234fb795f22ef781cdea075/pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go#L179
Like this no need to have another goroutine

Reply:
`EnsureLeaderElectionRuns` is not  the function used to start the leader election and is not blocking

Reply:
we shouldn't rely on it to start collecting the metric